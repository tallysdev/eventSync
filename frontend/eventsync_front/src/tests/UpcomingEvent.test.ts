import { mount } from '@vue/test-utils'
import { describe, it, expect } from 'vitest'
import UpcomingEvents from '@/components/home/UpcomingEvent.vue'

describe('UpcomingEvents.vue', () => {
  const upcomingEvents = [
    {
      id: 1,
      name: 'Titulo para encher linguiça e preencher o espaço para mostrar o texto',
      date: 'Sex, 30 ago',
      time: '19:00 - 21:00',
      location: 'UFRN/Ceres - Caicó, RN',
      buttonText: 'Saiba mais'
    },
    {
      id: 2,
      name: 'Partida de Futebol',
      date: '15 de Julho',
      time: '16:00 - 18:00',
      location: 'Estádio ABC',
      buttonText: 'Saiba mais'
    },
    {
      id: 3,
      name: 'Peça de Teatro',
      date: '20 de Julho',
      time: '20:00 - 22:00',
      location: 'Teatro 123',
      buttonText: 'Saiba mais'
    }
  ]
  const wrapper = mount(UpcomingEvents, {
    props: { upcomingEvents }
  })

  it('renders correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders the correct number of upcoming events', () => {
    expect(wrapper.findAll('v-carousel-item')).toHaveLength(upcomingEvents.length)
  })
})
