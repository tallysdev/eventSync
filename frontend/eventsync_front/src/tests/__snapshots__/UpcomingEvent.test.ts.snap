// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UpcomingEvents.vue > renders correctly 1`] = `
"<section data-v-19c29f04="" class="d-block" upcomingevents="[object Object],[object Object],[object Object]">
  <v-row data-v-19c29f04="" justify="center">
    <v-col data-v-19c29f04="" cols="12" lg="10">
      <h2 data-v-19c29f04="" class="text-h4 text-start mb-6 font-weight-bold">Eventos próximos</h2>
      <div data-v-19c29f04="" class="text-center mt-10">
        <p data-v-19c29f04="" class="text-h5 primary--text">Nenhum evento próximo disponível.</p>
      </div>
    </v-col>
  </v-row>
</section>"
`;
